ros_buildfarm/templates/release/rpm/sync_packages_to_main_job.xml.em

```diff
@@ -55,6 +55,23 @@
         ' --invalidate-expression "^ros-%s-.*"' % rosdistro_name,
         'echo "# END SECTION"',
     ]),
+))@
+@(SNIPPET(
+    'builder_shell',
+    script='\n'.join([
+        'echo "# BEGIN SECTION: mirror main repository content to disk"',
+    ] + [
+        'rsync --recursive --times --delete --itemize-changes rsync://127.0.0.1:1234/ros-main-%s-%s-SRPMS/ /var/repos/%s/main/%s/SRPMS/' % (os_name, os_code_name, os_name, os_code_name)
+        for os_name, os_code_name in set((os_name, os_code_name) for os_name, os_code_name, _ in sync_targets)
+    ] + [
+        'rsync --recursive --times --delete --exclude=debug --itemize-changes rsync://127.0.0.1:1234/ros-main-%s-%s-%s/ /var/repos/%s/main/%s/%s/' % (os_name, os_code_name, arch, os_name, os_code_name, arch)
```

Comment:
will the `--exclude=debug` cause any problems for a package named `ros-rolling-debug` or similar?

Reply:
The pattern matches path atoms, so it would only match a file or directory literally named "debug". To match your example, the pattern would need to be `--exclude=*debug`.

I tested this behavior _extensively_ to implement it in `pulp_rsync` the same way that `rsync` does it.

Really, this `--exclude` pattern isn't for excluding any content from being pulled out of plup, it's being used to make the local rsync operation leave the `debug` directory alone and not delete it for being absent in the upstream source.

Reply:
Thanks for clarifying. Makes sense to me :+1: